src/QAT/qat/purr/integrations/tket.py

```diff
@@ -366,12 +366,16 @@ def run_tket_optimizations(qasm_string, opts, hardware: QuantumHardwareModel) ->
         )
         return qasm_string
 
-    couplings = hardware.qubit_direction_couplings
+    couplings = hardware.qubit_direction_couplings.copy()
     optimizations_failed = False
     architecture = None
     if any(couplings):
         # Without default remapping pass multi-qubit gates don't get moved around, so
         # trying to apply them to a limited subset of qubits provides no value.
+        logical_qubit_map = {q.index: i for i, q in enumerate(hardware.qubits)}
+        for c in couplings:
+            control_ind, targer_ind = c.direction
```

Comment:
you mean target ?

Reply:
also, maybe they're just names but I advise against assigning control and target semantics to the tuple as you don't know which is which

Reply:
You do know which i which, in the sense that the coupling quality is based on the specific direction. Will fix typo.

Reply:
> You do know which is which, in the sense that the coupling quality is based on the specific direction

Oh I mean logically speaking, control and target semantics usually come from the circuit. I guess it's also fine with hw arch ...